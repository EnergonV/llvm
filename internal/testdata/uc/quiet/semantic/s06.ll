; ModuleID = 's06.c'
target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

; Function Attrs: nounwind uwtable
define i32 @main() #0 {
  %x = alloca i8, align 1
  %y = alloca i32, align 4
  %1 = load i32, i32* %y, align 4
  %2 = trunc i32 %1 to i8
  store i8 %2, i8* %x, align 1
  %3 = load i8, i8* %x, align 1
  %4 = sext i8 %3 to i32
  store i32 %4, i32* %y, align 4
  ret i32 0
}

attributes #0 = { nounwind uwtable "disable-tail-calls"="false" "less-precise-fpmad"="false" "no-frame-pointer-elim"="true" "no-frame-pointer-elim-non-leaf" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+sse,+sse2" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.ident = !{!0}

!0 = !{!"clang version 3.7.1 (tags/RELEASE_371/final)"}
