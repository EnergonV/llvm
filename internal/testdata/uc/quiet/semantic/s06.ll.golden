target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"
define i32 @main() {
0:
	%x = alloca i8
	%y = alloca i32
	%1 = load i32, i32* %y
	%2 = trunc i32 %1 to i8
	store i8 %2, i8* %x
	%3 = load i8, i8* %x
	%4 = sext i8 %3 to i32
	store i32 %4, i32* %y
	ret i32 0
}
